tests/functional_tests/functional_tests.py

```diff
@@ -141,11 +227,30 @@ def test_deserializing_json(self):
             '197bac7a044af33332865b9266e26d493bdd668a660e44d88ce1a998c23dbd67'
         )
 
-    def test_serializing_deserializing_json(self):
+    def test_deserializing_json_v2(self):
+        m = Macaroon.deserialize(
+            '{"l": "http://mybank/", "i": "we used our secret key", "s": '
+            '"197bac7a044af33332"'
+            ', "c": [{"l": null, "i": "test = caveat", "v": null}]}',
+            serializer=JsonSerializer()
+        )
+        assert_equal(
+            m.signature_bytes,
+            binascii.hexlify(b'197bac7a044af33332')
+        )
+
+    def test_serializing_deserializing_json_v1(self):
+        self._serializing_deserializing_json_with_version(MACAROON_V1)
```

Comment:
Same here, I'd rather just use the tooling pytest gives you for free for parameterization.